front/src/features/books/booksSlice.js

```diff
@@ -61,19 +51,20 @@ export const {
   setData,
 } = booksSlice.actions;
 
-export const fetchBooks = (pageNumber) => async (dispatch) => {
+export const fetchBooks = (pageNumber) => (dispatch) => {
   dispatch(fetchBooksDataStarted());
 
   try {
-    const data = await api.books.fetchBooks(pageNumber);
-    dispatch(setData(data));
-
-    dispatch(fetchBooksDataDataSucceeded());
+    api.books.fetchBooks(pageNumber, dispatch, fetchBooksDataDataSucceeded);
```

Comment:
j.w.

const data = api.fetchBooks(pageNumber)
dispatch(fetchBooksDataDataSucceeded(data))

po kolei, bez zbednego przekazywania funkcji jako argumenty. To ma w 99% przypadkow sens tylko przy jakichś callbackach w propsach w komponentach (np. onClick)

Reply:
Done.